specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/machineLearningServices.json

```diff
@@ -349,6 +349,48 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/listNotebookKeys": {
+      "get": {
```

Comment:
Thanks. I will try to explain why we want it to be a GET, not POST. A data scientist needs this notebook key to initialize a notebook component, but usually those data scientist only have reader access of the control plane, so if we make it a POST call, admin needs to grant data scientist contributor role to access the key, which is not ideal for the customer. I checked with the Azure Notebook team if they have any security concerns about making the listNotebookKeys a GET instead of POST, and they don't have any concern. So I'm wondering if ARM can make an exception for this API call?

Reply:
The HTTP verb and RBAC are separate.  The verb for this must be a POST to comply with ARM standards.  If this isn't actually a secret and users with only /read should be able to see it then you can use an authorizationActionMapping in your manifest to change which RBAC right is needed to call the API.  See https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderRegistration.html#authorizationactionmappings-optional